vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java

```diff
@@ -1146,7 +1146,9 @@ else if (w instanceof Dialog) {
       if (c != null) {
         p.mySetType(c.getType());
         if (p instanceof Decorator) {
-          ((Decorator) p).mySetState(c.getState());
+          if (!(p instanceof PropertySheet)) {
+            ((Decorator) p).mySetState(c.getState());
+          }
```

Comment:
There is a problem with this approach. The question is, why does this work for all other traits? The specific problem I found is that if you remove a row from the property sheet, then all of the values get shifter up. Try creating 3 rows x, y and z and give them values 1, 2 and 3. Test. Then edit and remove line y. Then test again and z now has the value of 2 instead of 3.

I think the problem is in the Propertysheet, not the PieceDefiner. The 'shape' of the ProprtySheet trait is dynamic, it depends on the number and order of the defined rows.  I think the bug is in PropertySheet.Ed.getState() which is blasting away the existing state and returning a default set of null values. This should be returning the actual state recorded in the pre-edit piece, less deleted rows + added rows.